libs/common/src/sql-db-entities/base-entities/auditable.base-entity.ts

```diff
@@ -1,13 +1,39 @@
-import { Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
-import { DbAuditUpdatedByUserId } from '../helpers';
+import { Column, UpdateDateColumn } from 'typeorm';
+import { DbRequestSource } from '../helpers';
 
 export abstract class AuditableBaseEntity {
-  @CreateDateColumn()
-  createdDate!: Date;
-
   @UpdateDateColumn()
-  updatedDate!: Date;
+  lastUpdatedAt!: Date;
+
+  @Column({ type: 'nvarchar', nullable: true })
+  lastUpdatedByPortalUserId!: string | null;
+
+  @Column({ type: 'nvarchar', nullable: true })
+  lastUpdatedByTfmUserId!: string | null;
+
+  @Column()
+  lastUpdatedByIsSystemUser!: boolean;
 
-  @Column({ type: 'nvarchar' })
-  updatedByUserId!: DbAuditUpdatedByUserId;
+  public updateActivityDetails(requestSource: DbRequestSource): void {
+    const { platform } = requestSource;
+    switch (platform) {
```

Comment:
You need to destructure it if you want to avoid a typescript error in the `default` case.

Originally, I wanted to have an error message that says "I don't recognise this platform: ${platform}". You get a tsc error if you do that using `requestSource.platform`, because it thinks `requestSource` is of type `never` which means it has no properties on it. If you destructure at the start, and then use `platform` in the `default` case, you only get a lint error which we have turned off in other services. In the implementation I currently have you are right that there is no point to doing that, but I'm going to update the error message and then there will be :)